fpdf/image_parsing.py

```diff
@@ -527,13 +532,133 @@ def __getitem__(self, tag):
     return newimgio.read(length)
 
 
+def _to_lzwdata(img, remove_slice=None, select_slice=None):
+    CLEAR_TABLE_MARKER = 256  # Special code to indicate table reset
+    EOD_MARKER = 257  # End-of-data marker
+    INITIAL_BITS_PER_CODE = 9  # Initial code bit width
+    MAX_BITS_PER_CODE = 12  # Maximum code bit width
+
+    data = bytearray(img.tobytes())
+
+    if remove_slice:
+        del data[remove_slice]
+    if select_slice:
+        data = data[select_slice]
+
+    if img.mode == "1":
+        row_size = ceil(img.size[0] / 8)
+        data_with_padding = data
+    else:
+        channels_count = len(data) // (img.size[0] * img.size[1])
+        row_size = img.size[0] * channels_count
+        data_with_padding = bytearray()
+        for i in range(0, len(data), row_size):
+            data_with_padding.extend(b"\0")
+            data_with_padding.extend(data[i : i + row_size])
+
+    data = data_with_padding
+
+    def clear_table():
+        """
+        Reset the encoding table and coding state to initial conditions.
+
+        """
+
+        table = {bytes([i]): i for i in range(256)}
+        next_code = EOD_MARKER + 1
+        bits_per_code = INITIAL_BITS_PER_CODE
+        max_code_value = (1 << bits_per_code) - 1
+        return table, next_code, bits_per_code, max_code_value
+
+    def pack_codes_into_bytes(codes):
```

Comment:
In order to conform to the existing code style on `fpdf2`, it's better to define those functions at the module level, an not inside the `_to_lzwdata()` function